packages/vue-app/template/mixins/fetch.client.js

```diff
@@ -0,0 +1,53 @@
+import Vue from 'vue'
+import { hasFetch } from '../utils'
+
+const isSsrHydration = (vm) => vm.$vnode && vm.$vnode.elm && vm.$vnode.elm.dataset && vm.$vnode.elm.dataset.ssrKey
+const nuxtState = window.<%= globals.context %>
+
+export default {
+  beforeCreate () {
+    if (hasFetch(this)) {
+      Vue.util.defineReactive(this, '$isFetching', false)
+      Vue.util.defineReactive(this, '$fetchError', null)
+    }
+  },
+  created () {
+    if (hasFetch(this) && isSsrHydration(this)) {
+      // Hydrate component
+      this._hydrated = true
+      this.$isLoading = false
+      this._lastFetchAt = Date.now()
+
```

Comment:
@Atinux perhaps we should avoid using `_` for public properties we expect people to use (and are even part of usage examples). Maybe a per-component `$fetchState` with `{ hydrated, loading, timestamp }` -- wdyt? (Or even attach these properties to the `$fetch` function itself)

Reply:
Yes, I will change to have something like `$fetchState` and also introducing `<nuxt-fetch>` component to use it like https://github.com/posva/vue-promised